tools/lsp/common/rename_component.rs

```diff
@@ -919,18 +1339,626 @@ export enum EnumBar { bar }
+                (
+                    Url::from_file_path(test::test_file_name("user4.slint")).unwrap(),
+                    r#"
+import { Foo } from "source.slint";
+
+export { Foo as User4Fxx }
+                "#
+                    .to_string(),
+                ),
+            ]),
+            false,
+        );
+
+        let doc =
+            document_cache.get_document_by_path(&test::test_file_name("source.slint")).unwrap();
+
+        let identifiers = find_struct_identifiers(doc.node.as_ref().unwrap(), "Foo");
+        assert_eq!(identifiers.len(), 1);
+        let edit =
+            rename_identifier_from_declaration(&document_cache, &identifiers[0], "XxxYyyZzz")
+                .unwrap();
+
+        let edited_text = compile_test_changes(&document_cache, &edit, false);
+
+        for ed in &edited_text {
+            let ed_path = ed.url.to_file_path().unwrap();
+            if ed_path == test::main_test_file_name() {
+                assert!(ed.contents.contains("XxxYyyZzz"));
+                assert!(!ed.contents.contains("Foo"));
+                assert!(ed.contents.contains("UserComponent"));
+                assert!(ed.contents.contains("import { XxxYyyZzz as User3Fxx }"));
+                assert!(ed.contents.contains("import { User4Fxx }"));
+            } else if ed_path == test::test_file_name("source.slint") {
+                assert!(ed.contents.contains("export struct XxxYyyZzz {"));
+                assert!(!ed.contents.contains("Foo"));
+            } else if ed_path == test::test_file_name("user.slint") {
+                assert!(ed.contents.contains("{ XxxYyyZzz as Bar }"));
+                assert!(ed.contents.contains("property <Bar> user-component-prop"));
+                assert!(!ed.contents.contains("Foo"));
+            } else if ed_path == test::test_file_name("user2.slint") {
+                assert!(ed.contents.contains("import { XxxYyyZzz }"));
+                assert!(ed.contents.contains("member: XxxYyyZzz,"));
+            } else if ed_path == test::test_file_name("user3.slint") {
+                assert!(ed.contents.contains("import { XxxYyyZzz }"));
+                assert!(ed.contents.contains("export { XxxYyyZzz }"));
+            } else if ed_path == test::test_file_name("user4.slint") {
+                assert!(ed.contents.contains("import { XxxYyyZzz }"));
+                assert!(ed.contents.contains("export { XxxYyyZzz as User4Fxx }"));
+            } else {
+                unreachable!();
+            }
+        }
+    }
+
+    #[test]
+    fn test_rename_enum_from_definition_ok() {
+        let document_cache = test::compile_test_with_sources(
+            "fluent",
+            HashMap::from([(
+                Url::from_file_path(test::main_test_file_name()).unwrap(),
+                r#"
+    export { Foo }
+
+    enum Foo /* Foo 1 */ {
+        M1, M2,
+    }
+
+    enum Foo /* Foo 3 */ {
+        test,
+    }
+
+    component Baz {
+        in-out property <Foo> baz-prop: Foo.test;
```

Comment:
I don't see the test that check that the `Foo.test` is properly renamed.

Reply:
True, that will be fixed in the next push.